### DISCUSSION TITLE
Tools and CI Meeting - Date 6/24

### DISCUSSION THREAD

**Discussion by makubacki** [tianocore/edk2#5818] [General]
# Status

Scheduled for 6/24 @ 4:30pm Pacific Time

# Agenda

Meeting notes are inline.

1. @bcran: Enabling the PR process for edk2-platforms and other TianoCore repositories.
   - Working through some remaining issues such as auto assigning reviewers
   - Last meeting: Implement support for Code Owners first and then add reviewer support later.
     - Challenge: Syntax of maintainers.txt and Code Owners is different - difficult to sync
       - Python script implemented that will help (Mike Kinney)
       - Need Python helper script to create Code Owners from maintainers.txt
   - Other repos are simpler than edk2 (like edk2-platforms)
     - Are we okay simply deleting maintainers.txt and just use Code Owners right away? Bringing the other settings from edk2 there.
     - The problem of not supporting "reviewers" translates to edk2-platforms as well
     - General idea was to stabilize things in edk2 and take the settings to other repos
   - @mdkinney: Run an experiment with a member of `edk2-reviewers` (does not have write permission) in the Code Owners file and see if they are added to the PR without the ability to add push label
   - @bcran: Check with maintainers in edk2-platforms if they are okay with switching to Code Owners (dropping maintainers.txt) and report back next week.
2. @bcran: Status of the BaseTools C to Python conversion project.
   - Project has stalled
   - Guidance is to move forward with enhancements to the C code for now
   - Other notes:
     - Should audit and remove unused tools
     - Should make existing Python code more Pythonic
3. @Ashwin-Veeraiah: Basetools support for `SPEC` keyword in INF files, `Defines` Section does not work. 
   - Current documentation provided for using it breaks the build process.
   
   - First reaction in similar cases is to update the docs to match the implementation (tool) behavior (when implementation has been the same for a long time)
   - Use case: Wanted to reuse the value from the INF file and in the C source code.
     - Need this use case supported to modify C code behavior on the spec version
   - GenMake is taking DEFINEs and moving it to the makefile
   - @Ashwin-Veeraiah has code changes for the issue that can be submitted
     - PR should initially go to [`edk2-basetools`](https://github.com/tianocore/edk2-basetools) and then it will be carried to edk2 `BaseTools` after that
       - `edk2-basetools` uses GitHub issues in case an issue needs to be created there
4. Open regarding "Dismiss stale pull request approvals when new commits are pushed" in https://github.com/tianocore/edk2/pull/5738#issuecomment-2179093587.
   - Decision is to leave the setting as is (keep approvals on commit pushes)
   - We'd need GitHub merge queues for this to efficiently work in any case

# Todos From Previous Meeting

- Follow up on `impact:security` vs CVE with infosec group.
- BZ to GitHub issue transition for new issues.
  - Next step: Send RFC with the idea and details about how issues would be processed and handled
- Investigate GitHub merge queue as a replacement for Mergify.

# Groups.io meeting invite:

https://edk2.groups.io/g/devel/viewevent?repeatid=43269&eventid=2159616&calstart=2024-01-08

# Microsoft Teams meeting

Join on your computer or mobile app
[Click here to join the meeting](https://teams.microsoft.com/l/meetup-join/19%3ameeting_ZDI2ZDg4NmMtMjI1My00MzI5LWFmYjAtMGQyNjUzNTBjZGYw%40thread.v2/0?context=%7b%22Tid%22%3a%2272f988bf-86f1-41af-91ab-2d7cd011db47%22%2c%22Oid%22%3a%2223af6561-6e1c-450d-b917-d9d674eb3cb6%22%7d)
Or call in (audio only)
+1 425-616-0754,,330776518# United States, Seattle
Phone Conference ID: 330 776 518#
Find a local number | [Reset PIN](https://mysettings.lync.com/pstnconferencing)
[Learn More](https://aka.ms/JoinTeamsMeeting) | [Meeting options](https://teams.microsoft.com/meetingOptions/?organizerId=23af6561-6e1c-450d-b917-d9d674eb3cb6&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&threadId=19_meeting_ZDI2ZDg4NmMtMjI1My00MzI5LWFmYjAtMGQyNjUzNTBjZGYw@thread.v2&messageId=0&language=en-US)

**Comment by Ashwin-Veeraiah** [comment]
Add to discussion : Basetools support for `SPEC` keyword in INF files, `Defines` Section does not work. 
Current documentation provided for using it breaks the build process.
